graphql/subcourse/fields.ts

```diff
@@ -86,6 +87,16 @@ export class ExtendedFieldsSubcourseResolver {
                     { joinAfterStart: { equals: true }, lecture: { some: { start: { gt: new Date() } } } },
                 ],
             });
+            if (isSessionPupil(context)) {
+                const pupil = await getSessionPupil(context);
+                const pupilGrade = gradeAsInt(pupil.grade);
+                if (pupilGrade) {
+                    filters.push({
+                        AND: [{ minGrade: { lte: pupilGrade }, maxGrade: { gte: pupilGrade } }],
```

Comment:
Out of curiosity, what is the intention of this PR?
It looks like we are trying to narrow down the courses based on the pupli who is requesting to show only the ones that are interesting to them.

But isn't this filter returning all the courses not available for his grade?

```suggestion
                        AND: [{ minGrade: { gte: pupilGrade }, maxGrade: { lte: pupilGrade } }],
```